packages/swirl-components/src/components/swirl-tree-navigation-item/swirl-tree-navigation-item.css

```diff
@@ -1,78 +1,127 @@
   width: 100%;
-  height: 2.5rem;
-  padding-right: var(--s-space-16);
-  padding-left: calc(var(--s-space-16) + 1.5rem + var(--s-space-8));
-  flex-shrink: 0;
-  justify-content: flex-start;
+  list-style: none;
+  border-radius: var(--s-border-radius-sm);
+}
+
+/* Link styles */
+.tree-navigation-item__link {
+  display: inline-flex;
   align-items: center;
+  width: 100%;
+  height: 2.25rem;
+  padding: 0 var(--s-space-16) 0 var(--tree-nav-item-padding, var(--s-space-8));
+  margin-bottom: var(--s-space-2);
+  gap: calc(var(--s-space-4) - var(--s-space-2));
+
+  /* Reset button/link styles */
   border: none;
-  color: var(--s-text-default);
   background-color: transparent;
+  color: var(--s-text-default);
+  cursor: pointer;
+  text-decoration: none;
+
+  /* Typography */
   font: inherit;
   font-size: var(--s-font-size-sm);
+  font-weight: var(--tree-nav-item-label-weight, var(--s-font-weight-medium));
   line-height: var(--s-line-height-sm);
-  text-decoration: none;
-  cursor: pointer;
-  gap: calc(var(--s-space-8) - var(--s-space-2));
 
-  &:hover {
-    background-color: var(--s-surface-hovered);
-  }
+  /* Border radius */
+  border-radius: var(--s-border-radius-sm);
+}
+
+/* Link states */
+.tree-navigation-item__link:hover {
+  background-color: var(--s-background-hovered);
+}
 
-  &:active {
-    background-color: var(--s-surface-pressed);
-  }
+.tree-navigation-item__link:active {
+  background-color: var(--s-background-pressed);
+}
 
-  &:focus {
-    outline: none;
-  }
+.tree-navigation-item__link:focus {
+  outline: none;
+}
 
-  &:focus-visible {
-    & .tree-navigation-item__label {
-      border-radius: var(--s-border-radius-xs);
-      box-shadow: 0 0 0 0.125rem var(--s-focus-default);
-    }
-  }
+.tree-navigation-item__link:focus-visible .tree-navigation-item__label {
+  border-radius: var(--s-border-radius-xs);
+  box-shadow: 0 0 0 0.125rem var(--s-focus-default);
 }
 
-.tree-navigation-item--active {
+/* Active link styles */
+.tree-navigation-item__link--active {
   background-color: var(--s-surface-raised-default);
-  box-shadow: inset 0.25rem 0 0 0 var(--s-border-highlight);
+  color: var(--s-text-highlight);
+  --tree-nav-item-color: var(--s-text-highlight);
+  --tree-nav-item-label-weight: var(--s-font-weight-medium);
 }
 
-.tree-navigation-item--active:hover {
+.tree-navigation-item__link--active:hover {
   background-color: var(--s-surface-raised-hovered);
 }
 
-.tree-navigation-item--active:active {
+.tree-navigation-item__link--active:active {
   background-color: var(--s-surface-raised-pressed);
 }
 
-.tree-navigation-item--has-icon {
-  padding-left: var(--s-space-16);
+/* Layout components */
+.tree-navigation-item--expanded {
+  height: 100%;
+}
+
+.tree-navigation-item__content {
+  display: inline-flex;
+  align-items: center;
 }
 
+/* Icon styles */
 .tree-navigation-item__icon {
   display: inline-flex;
   flex-shrink: 0;
+  padding-right: var(--s-space-8);
 }
 
+.tree-navigation-item__toggle-icon {
```

Comment:
Suggestion: `flex-shrink: 0` so it doesn't  shrink and disappear with longer labels or small viewports.